File: public/merchant/models/analytics.js
import m from "mithril";

export var Analytics = {
  	Data: {
		  UnapprovedListingsCount: 0,
		  ListingsCount: 0,
		  UsersCount: 0
	},
	GetAnalytics: () => {
		return  m.request({
				method: 'GET',
				url: '/api/merchants/analytics',
			}).then(function(response) {
				console.log("analytics response: ", response)
				Analytics.Data = response;
			}).catch(function(error) {
				console.error(error);
		});
	}
}

File: models/analytics.go
package models

type Analytics struct {
	UnapprovedListingsCount int `json:"UnapprovedListingsCount" db:"unapproved_listings_count"`
	ListingsCount           int `json:"ListingsCount" db:"listings_count"`
	UsersCount              int `json:"UsersCount" db:"users_count"`
}
